src/index.js

```diff
@@ -37,6 +37,7 @@ function startsWith ( str, prefix ) {
 	return str.slice( 0, prefix.length ) === prefix;
 }
 
+const isCjsPromises = Object.create(null);
```

Comment:
My worry is this could be cache poisoning if this changes between runs (eg a file changes from ESM to CJS or vice versa).

Do you know what the root cause was on the stalling that this fixes it?

Reply:
I think it's because rollup won't call transform again for paths where code hasn't changed, so [.resolve](https://github.com/rollup/rollup-plugin-commonjs/blob/fd0d302178f529ef953730c8d97299ca7ab5a577/src/index.js#L118) will never be called. I'm unfortunately not familiar with the flow, but yes, I'm sure there's a better way.

Reply:
Ah thanks that makes complete sense to understand the cause.